Bug: Can't build on latest arch linux

**Describe the bug**
Can't build on latest arch linux after updating my system I just couldn't build it also several other libraries too .. not sure if it's arch or assimp .. any ways tested it with the latest arch:base-devel docker image and it doesn't compile .. it give this error message : 
```
FAILED: code/CMakeFiles/assimp.dir/AssetLib/FBX/FBXAnimation.cpp.o
/usr/bin/c++ -DASSIMP_BUILD_DLL_EXPORT -DASSIMP_BUILD_NO_C4D_IMPORTER -DASSIMP_BUILD_NO_M3D_EXPORTER -DASSIMP_BUILD_NO_M3D_IMPORTER -DASSIMP_IMPORTER_GLTF_USE_OPEN3DGC=1 -DMINIZ_USE_UNALIGNED_LOADS_AND_STORES=0 -DOPENDDLPARSER_BUILD -DRAPIDJSON_HAS_STDSTRING=1 -DRAPIDJSON_NOMEMBERITERATORCLASS -Dassimp_EXPORTS -I/home/thegeeko/repos/assimp/build/include -I/home/thegeeko/repos/assimp/build -I/home/thegeeko/repos/assimp/include -I/home/thegeeko/repos/assimp/code -I/home/thegeeko/repos/assimp/. -I/home/thegeeko/repos/assimp/contrib/zlib -I/home/thegeeko/repos/assimp/build/contrib/zlib -I/home/thegeeko/repos/assimp/code/../contrib/pugixml/src -I/home/thegeeko/repos/assimp/code/../contrib/rapidjson/include -I/home/thegeeko/repos/assimp/code/../contrib -I/home/thegeeko/repos/assimp/code/../contrib/unzip -I/home/thegeeko/repos/assimp/code/../contrib/openddlparser/include -I/home/thegeeko/repos/assimp/code/../include -I/home/thegeeko/repos/assimp/build/code/../include -I/home/thegeeko/repos/assimp/code/.. -fvisibility=hidden -fno-strict-aliasing -Wall  -std=gnu++17 -fPIC -Wall -Werror -MD -MT code/CMakeFiles/assimp.dir/AssetLib/FBX/FBXAnimation.cpp.o -MF code/CMakeFiles/assimp.dir/AssetLib/FBX/FBXAnimation.cpp.o.d -o code/CMakeFiles/assimp.dir/AssetLib/FBX/FBXAnimation.cpp.o -c /home/thegeeko/repos/assimp/code/AssetLib/FBX/FBXAnimation.cpp
/home/thegeeko/repos/assimp/code/AssetLib/FBX/FBXAnimation.cpp: In constructor ‘Assimp::FBX::AnimationCurve::AnimationCurve(uint64_t, const Assimp::FBX::Element&, const std::string&, const Assimp::FBX::Document&)’:
/home/thegeeko/repos/assimp/code/AssetLib/FBX/FBXAnimation.cpp:63:20: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
   63 |     const Element &KeyTime = GetRequiredElement(sc, "KeyTime");
      |                    ^~~~~~~
/home/thegeeko/repos/assimp/code/AssetLib/FBX/FBXAnimation.cpp:63:48: note: the temporary was destroyed at the end of the full expression ‘Assimp::FBX::GetRequiredElement((* & sc), std::__cxx11::basic_string<char>(((const char*)"KeyTime"), std::allocator<char>()), 0)’
   63 |     const Element &KeyTime = GetRequiredElement(sc, "KeyTime");
      |                              ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
/home/thegeeko/repos/assimp/code/AssetLib/FBX/FBXAnimation.cpp:64:20: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
   64 |     const Element &KeyValueFloat = GetRequiredElement(sc, "KeyValueFloat");
      |                    ^~~~~~~~~~~~~
/home/thegeeko/repos/assimp/code/AssetLib/FBX/FBXAnimation.cpp:64:54: note: the temporary was destroyed at the end of the full expression ‘Assimp::FBX::GetRequiredElement((* & sc), std::__cxx11::basic_string<char>(((const char*)"KeyValueFloat"), std::allocator<char>()), 0)’
   64 |     const Element &KeyValueFloat = GetRequiredElement(sc, "KeyValueFloat");
      |                                    ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

**To Reproduce**
Steps to reproduce the behavior:
1. docker pull archlinux:base-devel
2. docker run -i -t archlinux:base-devel /bin/bash
3. git clone --depth 1 https://github.com/assimp/assimp.git
4. mkdir build && cd build && cmake .. && make

**Platform (please complete the following information):**
 - OS: Linux 6.3.8-arch1-1 SMP PREEMPT_DYNAMIC +0000 x86_64 GNU/Linux
 - Version master

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -268,6 +268,11 @@ IF ((CMAKE_C_COMPILER_ID MATCHES "GNU") AND NOT MINGW)
     SET(CMAKE_CXX_STANDARD 17)
     SET(CMAKE_POSITION_INDEPENDENT_CODE ON)
   ENDIF()
+  
+  IF(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 13)
+    MESSAGE(STATUS "GCC13 detected disabling \"-Wdangling-reference\" in Cpp files as it appears to be a false positive")
+    ADD_COMPILE_OPTIONS("$<$<COMPILE_LANGUAGE:CXX>:-Wno-dangling-reference>")
+  ENDIF()
   # hide all not-exported symbols
   IF(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "mips64" )
     SET(CMAKE_CXX_FLAGS "-mxgot -fvisibility=hidden -fno-strict-aliasing -Wall ${CMAKE_CXX_FLAGS}")

